lib/modules/hosts/ireddit.js

```diff
@@ -0,0 +1,27 @@
+/* @flow */
+
+import { Host } from '../../core/host';
+import { DAY, string, Thing, getPostMetadata } from '../../utils';
+import { ajax } from '../../environment';
+
+export default new Host('ireddit', {
+	name: 'ireddit',
+	domains: ['i.redd.it'],
+	options: {
+		useiReddit: {
+			title: 'iredditUseIReddItTitle',
+			description: 'iredditUseIReddItDesc',
+			value: false,
+			type: 'boolean',
+		},
+	},
+	detect: (url, thing) => thing && thing.isPost() && thing.id(),
+	async handleLink(href, id) {
+		let postMetadata = await getPostMetadata({ id: id.replace('t3_', '') });
+		newhref = postMetadata.preview.images[0].source.url; //got the url!
```

Comment:
There needs to be two branches here, for static images and gifs (which will give us the .mp4). Something like:

```js
const preview = postMetadata.preview.images[0];
if (preview.variants.mp4) {
    return {
        type: 'VIDEO',
        controls: false,
        loop: true,
        muted: true,
        fallback: preview.variants.gif && preview.variants.gif.source.url,
        sources: [{
            source: preview.variants.mp4.source.url,
            type: 'video/mp4',
        }],
    };
} else {
    return {
        type: 'IMAGE',
        src: preview.source.url,
    };
}
```